### PR TITLE
New user man: First version

### LINKED ISSUE #709: Review and improve user man
Review and refactor the whole user man, in particular the core section to:
- make it use case driven 
- add references to tips and tutorials when relevant.

The doc is generated and hosted here: 
https://docs.taipy.io/en/feature-709-user-man/

### PR DESCRIPTION
**Pull request by jrobinAV**
Addresses #709 and [taipy-enterprise#241](https://github.com/Avaiga/taipy-enterprise/issues/241).

Temporarily deployed [here](https://docs.taipy.io/en/feature-709-user-man/manuals/new-user-manual/).

This is a proposal for a new user manual structure. The objective is to make it more usage-driven. 

## Notes for reviewers
This PR touches a lot of files. Too many. To review it, please generate the documentation and browse it.
    
image

## DONE
-  [x] The tabs have been removed, and all the top-level parts have been added to the navigation
-  [x] The Gui has been renamed and slightly changed.
    - [x] The new visual element page now encapsulates all controls and blocks together
    - [x] Page part has been moved to the second position
-  [x] The old core part has been split to exhibit the usages: 
    * Data integration,
    * Task orchestration,
    * What-if-analysis,
    * scenario and data management, 
    * version management, 
    * scheduler
- [x] The configuration part groups the GUI and core configurations.
- [x] The Data integration part contains everything related to data nodes.
- [x] The task orchestration contains everything related to submissions, DAGs, jobs, seqs
- [x] The what-if part contains everything related to what-if-analysis:
    - [x] multiple scenarios, 
    - [x] scopes and cycles, 
    - [x] comparators.
- [x] The scenario and data management part contains an exhaustive description of all entities' usage.
    - [x] Data nodes
    - [x] Scenario
    - [x] Task
    - [x] Sequence
    - [x] Cycle
    - [x] Job
    - [x] Submission
- [x] Review all transitions.
- [x] Review all the places where a config.py file must be imported and add a download link.
- [x] Review all cross-refs, code samples, and pictures.
- [x] Review all TODOs.
- [x] Review the old manual to list missing parts.
- [x] The old structure must be removed.

### CODE REVIEW

File: mkdocs.yml_template [hunk 1]
```diff
@@ -63,14 +63,130 @@ nav:
...
+
+    - "New User Manual":
+      - "About Taipy's User Manuals": manuals/new-user-manual/index.md
+      - "User interface":
+        - "User interface": manuals/new-user-manual/gui/index.md
+        - "Visual elements": manuals/new-user-manual/gui/controls.md
+        - "Variable binding": manuals/new-user-manual/gui/binding.md
+        - "Callbacks": manuals/new-user-manual/gui/callbacks.md
+        - "Pages": manuals/new-user-manual/gui/pages/index.md
+        - "Layout":
```
Comment (trgiangdo) [inline mkdocs.yml_template]: There is no index.md for the Layout page, so the arrow when built is a bit off.
image

You can see that the arrow for the "Layout" is at the end, while others' arrows are at the beginning

**Review by trgiangdo — COMMENTED** [review]
The `manuals/gui/viselements/blocks.md` is missing from the new userman.

I haven't fully checked all files yet, but I think we need to check for all existing pages to be migrated correctly to the new userman

**Comment by jrobinAV** [comment]
> The `manuals/gui/viselements/blocks.md` is missing from the new userman.
> 
> I haven't fully checked all files yet, but I think we need to check for all existing pages to be migrated correctly to the new userman

I propose moving it to the layout part since it concerns the page design and layout but not the content or interactivity like other controls.

File: mkdocs.yml_template (hunk 1 above)
Comment (jrobinAV) [reply mkdocs.yml_template]: yes. I should add an index.md page. I modified the todo list

**Comment by jrobinAV** [comment]
The "user interface" and "Data integration" parts are ready to be reviewed.

**Comment by FlorianJacta** [comment]
# Index User Man:

- "For each topic, we are trying to provide as many examples as possible so that you as a Taipy user can perform a specific task." I don't see the use of this sentence.

- Why put the "Supported Python versions" section here? I want to see the quick access, and there are a lot of texts before viewing it that I don't find so important.

- "Graphic User Interface" > Graphical User Interface

After the quick access, I found the page complicated for no reason, but it may be what it must be.

## User interface index

I don't find this page interesting. It is between an overview and a deep dive into technicalities that are not useful and explained elsewhere in the doc. This could be frightening as it is the beginning of the User Man for the User Interface.

## Visual Elements index page

I think there is too much text before seeing the list of controls. I want to see the controls.
And each control takes too much place on the page.


## Datanode integration

Have a graph for "Why use Data Nodes" instead. This is really hard to read and doesn't highlight the advantages.

### Data Node config

"data_node_cfg = Config.configure_data_node(id="data_node_cfg")" > data_node_cfg = Config.configure_data_node(id="data_node")

I don't think it should be "_cfg" for the id. Also, we could maybe have a code sample somewhere that shows the config and the creation of the data node. I don't know where to put it but this is the basic example of "Data Integration" and it is not shown anywhere.


### Data node history and validity

I found the code of "Edits" not so clear. Maybe because it is not an actual use case or because the formatting makes it hard to understand


---

This User Manual is really much better!

**Comment by jrobinAV** [comment]
> # Index User Man:
> * "For each topic, we are trying to provide as many examples as possible so that you as a Taipy user can perform a specific task." I don't see the use of this sentence.
> 
> Why put the "Supported Python versions" section here? I want to see the quick access, and there are a lot of texts before viewing it that I don't find so important.
> 

This is not covered by the current PR. I did not change the content of the user-man/index.md.
However, I agree with your comments.

**Comment by jrobinAV** [comment]
> * "Graphic User Interface" > Graphical User Interface

OK

> After the quick access, I found the page complicated for no reason, but it may be what it must be.

OK, let me discuss that with Fabien.

> ## User interface index
> I don't find this page interesting. It is between an overview and a deep dive into technicalities that are not useful and explained elsewhere in the doc. This could be frightening as it is the beginning of the User Man for the User Interface.
>

OK, let me discuss that with Fabien.

> ## Visual Elements index page
> I think there is too much text before seeing the list of controls. I want to see the controls. And each control takes too much place on the page.
> 

I agree. Let me discuss that with Fabien.

**Comment by jrobinAV** [comment]
> ## Datanode integration
> Have a graph for "Why use Data Nodes" instead. This is really hard to read and doesn't highlight the advantages.
> 

OK. Do you mean a graphic instead of a graph? Do you have any idea? 
I won't cover this feedback in this current PR.


> ### Data Node config
> "data_node_cfg = Config.configure_data_node(id="data_node_cfg")" > data_node_cfg = Config.configure_data_node(id="data_node")
> 
> I don't think it should be "_cfg" for the id. Also, we could maybe have a code sample somewhere that shows the config and the creation of the data node. I don't know where to put it but this is the basic example of "Data Integration" and it is not shown anywhere.
> 

OK. I agree for the `_cfg`. Let me think about the code sample topic.


> ### Data node history and validity
> I found the code of "Edits" not so clear. Maybe because it is not an actual use case or because the formatting makes it hard to understand
> 

hum... Any specific drawback you can specify so we can improve it?

**Comment by jrobinAV** [comment]
The task orchestration part is ready for review.

**Comment by AlexandreSajus** [comment]
I reviewed the structure, data integration, and task integration parts. No issues on my side. The structure is definitely a huge upgrade. It is way easier to find what you need according to your use case!

**Comment by joaoandre-avaiga** [comment]
I've reviewed data-integration and configuration sections. Everything seems to be in order except the links to the user manual, as you already mencioned on the email.

File: mkdocs.yml_template [hunk 2]
```diff
@@ -64,115 +65,136 @@ nav:
...
+      - "User interface":
+        - "User interface": manuals/userman/gui/index.md
+        - "Visual elements":
+          - "Visual elements": manuals/userman/gui/viselements/index.md
+          - "Introduction": manuals/userman/gui/viselements/introduction.md
+          - "Styling":
+              - "Styling": manuals/userman/gui/styling/index.md
+              - "The Stylekit": manuals/userman/gui/styling/stylekit.md
+        - "Pages": manuals/userman/gui/pages/index.md
+        - "Variable binding": manuals/userman/gui/binding.md
```
Comment (FabienLelaquais) [inline mkdocs.yml_template]:
```suggestion
        - "Binding variables": manuals/userman/gui/binding.md
```

File: mkdocs.yml_template [hunk 3]
```diff
@@ -64,115 +65,136 @@ nav:
...
+        - "Job":
+          - "Job": manuals/userman/sdm/job/index.md
+        - "Sequence":
+          - "Sequence": manuals/userman/sdm/sequence/index.md
+        - "Cycle":
+          - "Cycle": manuals/userman/sdm/cycle/index.md
+
+      - "Configuration":
+        - "Configuration": manuals/userman/configuration/index.md
+        - "Configuring GUI service": manuals/userman/configuration/gui-config.md
```
Comment (FabienLelaquais) [inline mkdocs.yml_template]:
```suggestion
        - "Configuring the GUI service": manuals/userman/configuration/gui-config.md
```
There's more than the GUI service in there. So we must talk.

File: mkdocs.yml_template [hunk 4]
```diff
@@ -64,115 +65,136 @@ nav:
...
+        - "Configuring Core service": manuals/userman/configuration/core-config.md
```
Comment (FabienLelaquais) [inline mkdocs.yml_template]:
```suggestion
        - "Configuring the Core service": manuals/userman/configuration/core-config.md
```

File: mkdocs.yml_template [hunk 5]
```diff
@@ -64,115 +65,136 @@ nav:
...
+        - "Configuring job execution ": manuals/userman/configuration/job-config.md
```
Comment (FabienLelaquais) [inline mkdocs.yml_template]:
```suggestion
        - "Configuring jobs execution": manuals/userman/configuration/job-config.md
```

File: mkdocs.yml_template [hunk 6]
```diff
@@ -64,115 +65,136 @@ nav:
...
+        - "Config checker": manuals/userman/configuration/config-checker.md
```
Comment (FabienLelaquais) [inline mkdocs.yml_template]:
```suggestion
        - "Configuration checker": manuals/userman/configuration/config-checker.md
```

**Review by FabienLelaquais — APPROVED** [review]
Excellent, great, amazing job there!

**Review by toan-quach — APPROVED** [review]
lgtm!

File: docs/manuals/userman/what-if-analysis/index.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
+In this section, we explore how to use *scenarios* for what-if analysis in Taipy.
+What-if analysis is all about exploring different scenarios to see how variable
+changes affect outcomes. It's essentially asking "What if?" questions and examining
+the results.
+
+# What is a Scenario?
+A `Scenario^` in Taipy represents an instance of a business problem. Among other features
+scenarios own data nodes. Two different scenarios can own two alternative data nodes
```
Comment (trgiangdo) [inline docs/manuals/userman/what-if-analysis/index.md]: The sentence doesn't make sense to me. Is this what you mean?
```suggestion
A `Scenario^` in Taipy represents an instance of a business problem.
Scenarios own data nodes, and two different scenarios can own two alternative data nodes
```

File: docs/manuals/userman/what-if-analysis/index.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+(from the same `DataNodeConfig^`). This is particularly handy when you want to represent
+different situation, assumption or condition, and compare the data between them.
```
Comment (trgiangdo) [inline docs/manuals/userman/what-if-analysis/index.md]:
```suggestion
(from the same `DataNodeConfig^`). This is particularly handy when you want to represent
different situations, assumptions or conditions, and compare the data between them.
```

File: docs/manuals/userman/what-if-analysis/index.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+    For more details, see the [scenario comparison](scenario-comparison.md) page.
+
+# How to use Scenarios?
+
+What-if analysis in Taipy involves configuring *data nodes*, setting up *scenario* configurations,
+and instantiating new *scenarios* at runtime. Each *scenario* holding an alternative data node,
+represents a different situation, assumption or condition, and as an end-user you can compare
+the different scenarios to understand the impact of various decisions, or events on your data.
+
+Here is the main principle:
```
Comment (trgiangdo) [inline docs/manuals/userman/what-if-analysis/index.md]:
```suggestion
Here are the main principles:
```
or
```suggestion
Here are the main steps:
```

File: docs/manuals/userman/what-if-analysis/index.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+1. **Configuring data nodes**: The first step consist in configuring the data nodes that will be
```
Comment (trgiangdo) [inline docs/manuals/userman/what-if-analysis/index.md]:
```suggestion
1. **Configuring data nodes**: The first step consists in configuring the data nodes that will be
```

File: docs/manuals/userman/what-if-analysis/scenario-comparison.md [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+include-markdown "./code-example/scenario-comparison/comparison-example.py"
+comments=false
+%}
+```
+Here is the
+<a href="./code-example/scenario-comparison/comparison-example.py" download>complete python code</a>
+corresponding to the example.
+
+
+## User define function
```
Comment (trgiangdo) [inline docs/manuals/userman/what-if-analysis/scenario-comparison.md]: I would rename the `User defined function` as `Comparator function`, also in the paragraph. Then we can clarify that the comparator function should be defined by the developer.

File: mkdocs.yml_template (hunk 3 above)
Comment (jrobinAV) [reply mkdocs.yml_template]: @FabienLelaquais , Should we rename this to "Configuring the Gui"?

File: docs/manuals/userman/what-if-analysis/scenario-comparison.md (hunk 11 above)
Comment (jrobinAV) [reply docs/manuals/userman/what-if-analysis/scenario-comparison.md]: Yep

**Comment by jrobinAV** [comment]
I got most of the feedback. I am merging the PR in develop. Please feel free to continue the review on develop branch/version.

Thank you for your help.

**Pull request merged**